### DISCUSSION TITLE
Static Caching with Cart

### DISCUSSION THREAD

**Discussion by sattyframework** [duncanmcclean/simple-commerce#974] [Help] [answered]
Hey there, wondered if I could get some guidance on how to utilise static caching but keep dynamic content. 
For the most part I have been able to work around this by making API calls to get the cart and then loading the cart details in a Vue component. 
However I have reached a sticking point. I am getting the cart by passing {{ sc:cart:id }} into my vue component where I make the API call. However cart id is caching now too. 

For example when I complete checkout on an order and clear the cart, it does start a new cart however the {{ sc:cart:id }} is cached and is returning the previous cart. The issue resolves after clearing static caching. I have tried using statamic's {{ no_cache }} tag but I find this quite intermittent.

So my question is, is it possible to retrieve the current cart from my controller when I make the API call, rather than passing it through my vue template? 

Any help is appreciated, thanks.

**Comment by duncanmcclean** [comment, marked as answer]
> So my question is, is it possible to retrieve the current cart from my controller when I make the API call, rather than passing it through my vue template?

Yes, there's a "cart index" endpoint you can use to get the customer's current cart. The URL is `/!/simple-commerce/cart`

**Reply by sattyframework** [reply]
Ahh amazing, thank youu!
